src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/InstanceRepository.cs

```diff
@@ -62,234 +57,155 @@ public InstanceRepository(IOptions<AzureCosmosSettings> cosmosettings)
         {
-            try
-            {
-                string instanceOwnerIdString = instanceOwnerId.ToString();
+            string instanceOwnerIdString = instanceOwnerId.ToString();
 
-                FeedOptions feedOptions = new FeedOptions
-                {
-                    PartitionKey = new PartitionKey(instanceOwnerIdString),
-                    MaxItemCount = 100,
-                };
+            FeedOptions feedOptions = new FeedOptions
+            {
+                PartitionKey = new PartitionKey(instanceOwnerIdString),
+                MaxItemCount = 100,
+            };
 
-                IQueryable<Instance> filter = _client
-                    .CreateDocumentQuery<Instance>(_collectionUri, feedOptions)
-                    .Where(i => i.InstanceOwnerId == instanceOwnerIdString);
+            IQueryable<Instance> filter = _client
+                .CreateDocumentQuery<Instance>(_collectionUri, feedOptions)
+                .Where(i => i.InstanceOwnerId == instanceOwnerIdString);
 
-                IDocumentQuery<Instance> query = filter.AsDocumentQuery<Instance>();
+            IDocumentQuery<Instance> query = filter.AsDocumentQuery<Instance>();
 
-                FeedResponse<Instance> feedResponse = await query.ExecuteNextAsync<Instance>();
+            FeedResponse<Instance> feedResponse = await query.ExecuteNextAsync<Instance>();
 
-                return feedResponse.ToList<Instance>();
-            }
-            catch (DocumentClientException e)
-            {
-                if (e.StatusCode == HttpStatusCode.NotFound)
-                {
-                    return null;
-                }
-                else
-                {
-                    throw;
-                }
-            }
+            return PostProcess(feedResponse.ToList<Instance>());
         }
 
-        /// <summary>
-        /// Update instance for a given form id
-        /// </summary>
-        /// <param name="instanceId">the instance id</param>
-        /// <param name="item">the instance</param>
-        /// <returns>The instance</returns>
-        public async Task<Instance> UpdateInstanceInCollectionAsync(Guid instanceId, Instance item)
+        /// <inheritdoc/>
+        public async Task<Instance> Update(Instance item)
         {
-            ResourceResponse<Document> createDocumentResponse = await _client.ReplaceDocumentAsync(UriFactory.CreateDocumentUri(databaseId, collectionId, instanceId.ToString()), item);
+            PreProcess(item);
+
+            ResourceResponse<Document> createDocumentResponse = await _client
+                .ReplaceDocumentAsync(UriFactory.CreateDocumentUri(databaseId, collectionId, item.Id), item);
             Document document = createDocumentResponse.Resource;
             Instance instance = JsonConvert.DeserializeObject<Instance>(document.ToString());
 
+            return PostProcess(instance);
+        }
+
+        private void PreProcess(Instance instance)
```

Comment:
Why do we need this void method? What does it actually do?

Reply:
Have added description and reworked logic.